cores/esp8266/WString.cpp

```diff
@@ -313,6 +339,16 @@ unsigned char String::concat(unsigned long num) {
     return concat(buf, strlen(buf));
 }
 
+unsigned char String::concat(long long num) {
+    char buf[2 + 3 * sizeof(long long)];
+    return concat(buf, sprintf(buf, "%lld", num));
+}
+
+unsigned char String::concat(unsigned long long num) {
+    char buf[1 + 3 * sizeof(unsigned long long)];
+    return concat(buf, sprintf(buf, "%llu", num));
+}
+
```

Comment:
`4 * sizeof(...)`?

Reply:
In this case, base is 10.
So it's `ceil(log(2^(s*8))/log(10)/s)` with `s=sizeof(type)`.
Result is 3 even with `long long`.

Reply:
Ah, it is base10... So, basically:
```python
>>> len(str(1 << 63))
19
```
Plus sign, plus `\0`

I remembered the to_string sources I looked some time ago, all of to_string the types used 'times 4', I'd guess for consistency and to avoid signed / unsigned slight differences

Reply:
I kept 3 for coherency with above/below lines. Indeed 4 is used when base is a variable.